arangod/RocksDBEngine/Listeners/RocksDBShaCalculator.cpp

```diff
@@ -269,11 +269,19 @@ RocksDBShaCalculator::RocksDBShaCalculator(application_features::ApplicationServ
 
 /// @brief Shutdown the background thread only if it was ever started
 RocksDBShaCalculator::~RocksDBShaCalculator() {
+  waitForShutdown();
```

Comment:
agree with the suggestion of only checking if shaThread is still running, as waitForShutDown() will be called before the destructor in RocksDBShaCalculator::stop()
suggestion:
`TRI_ASSERT(!_shaThread.isRunning())`